### ISSUE TITLE
[5.4.12] Preloader HarmonyInteropFix is throwing Exception.

### ISSUE THREAD

**Issue by BlazingTwist** [BepInEx/BepInEx#279]
**Describe the issue**
When using the 5.4.12 BepInEx.Preloader.dll while starting a game, an exception will be thrown.
The Exception does not occur when using the 5.4.11 BepInEx.Preloader.dll

**To Reproduce**
Start a x64, mono, Unity game after a fresh install of BepInEx 5.4.12.
-> Preloader throws an Exception, `plugins` and `patchers` directory isn't created.

**Expected behavior**
Preloader does not throw an Exception, `plugins` and `patchers` directory are created, mods load normally.

**Screenshots and logs**
preloader.log:
```

HarmonyLib.HarmonyException: IL Compile Error (unknown location) ---> System.InvalidProgramException: Invalid IL code in DMD<LoadFile>?1992125184:_BepInEx_Preloader_RuntimeFixes_HarmonyInteropFix::LoadFile (string): IL_0008: ret       


  at (wrapper managed-to-native) System.RuntimeMethodHandle.GetFunctionPointer(intptr)
  at System.RuntimeMethodHandle.GetFunctionPointer () [0x00000] in <44afb4564e9347cf99a1865351ea8f4a>:0 
  at MonoMod.RuntimeDetour.Platforms.DetourRuntimeILPlatform.GetFunctionPointer (System.Reflection.MethodBase method, System.RuntimeMethodHandle handle) [0x00000] in <45d5519965ed40d2af73eecdf7fbc188>:0 
  at MonoMod.RuntimeDetour.Platforms.DetourRuntimeILPlatform.GetNativeStart (System.Reflection.MethodBase method) [0x0004d] in <45d5519965ed40d2af73eecdf7fbc188>:0 
  at MonoMod.RuntimeDetour.DetourHelper.GetNativeStart (System.Reflection.MethodBase method) [0x00005] in <45d5519965ed40d2af73eecdf7fbc188>:0 
  at MonoMod.RuntimeDetour.Detour._TopApply () [0x00025] in <45d5519965ed40d2af73eecdf7fbc188>:0 
  at MonoMod.RuntimeDetour.Detour._RefreshChain (System.Reflection.MethodBase method) [0x00149] in <45d5519965ed40d2af73eecdf7fbc188>:0 
  at MonoMod.RuntimeDetour.Detour.Apply () [0x00053] in <45d5519965ed40d2af73eecdf7fbc188>:0 
  at MonoMod.RuntimeDetour.Detour..ctor (System.Reflection.MethodBase from, System.Reflection.MethodBase to, MonoMod.RuntimeDetour.DetourConfig& config) [0x002e1] in <45d5519965ed40d2af73eecdf7fbc188>:0 
  at DMD<Refresh>?1992125184._MonoMod_RuntimeDetour_ILHook+Context::Refresh (MonoMod.RuntimeDetour.ILHook+Context this) [0x00128] in <107811febea9422db03f152b2848fe07>:0 
  at DMD<>?1992125184.Trampoline<MonoMod.RuntimeDetour.ILHook+Context::Refresh>?-810517504 (System.Object ) [0x00020] in <04baeadfca7c4af788ed74700cf307c7>:0 
  at HarmonyLib.Internal.RuntimeFixes.StackTraceFixes.OnILChainRefresh (System.Object self) [0x00000] in <a6dd467dce664b88938cb422abccb8df>:0 
  at MonoMod.RuntimeDetour.ILHook.Apply () [0x00059] in <45d5519965ed40d2af73eecdf7fbc188>:0 
  at HarmonyLib.PatchFunctions.ReversePatch (HarmonyLib.HarmonyMethod standin, System.Reflection.MethodBase original, System.Reflection.MethodInfo postTranspiler, System.Reflection.MethodInfo postManipulator) [0x00158] in <a6dd467dce664b88938cb422abccb8df>:0 
   --- End of inner exception stack trace ---
  at HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) [0x00045] in <a6dd467dce664b88938cb422abccb8df>:0 
  at HarmonyLib.PatchClassProcessor.Patch () [0x00084] in <a6dd467dce664b88938cb422abccb8df>:0 
  at HarmonyLib.Harmony.PatchAll (System.Type type) [0x00008] in <a6dd467dce664b88938cb422abccb8df>:0 
  at HarmonyLib.Harmony.CreateAndPatchAll (System.Type type, System.String harmonyInstanceId) [0x0001e] in <a6dd467dce664b88938cb422abccb8df>:0 
  at BepInEx.Preloader.RuntimeFixes.HarmonyInteropFix.Apply () [0x0000a] in <2dd5991144a54f5db40ffaa2e3b971d1>:0 
  at BepInEx.Preloader.Preloader.Run () [0x00005] in <2dd5991144a54f5db40ffaa2e3b971d1>:0 
```

**Desktop (please complete the following information):**
Streets of Rogue v95, x64, mono
BepInEx 5.4.12
Unity 2019.4.28f1

**Additional context**
Notably 5.4.12 x86 as well as x64 did not throw the exception when testing with a handful of different Unity games.

**Comment by TheOneOgre** [comment]
Having the exact same issue using BepInEx 5.4.12 with Dyson Sphere Program. 

Version 5.4.11 works properly, whilst .12 does not.

**Comment by ghorsington** [comment]
This is now fixed in 5.4.13
